lib/github/middleware.js

```diff
@@ -105,4 +106,56 @@ module.exports = function middleware(callback) {
       }
     }
   });
+}
+
+
+function workerMiddleware(callbackName) {
+  return withSentry(async (context) => {
+    enhanceOctokit(context.github, context.log);
+
+    newrelic.setControllerName(`github/events.${context.name}`);
+    let webhookEvent = context.name;
+    if (context.payload.action) {
+      webhookEvent = `${webhookEvent}.${context.payload.action}`;
+    }
+    newrelic.addCustomAttributes({
+      'Webhook ID': context.id,
+      'Webhook Event': webhookEvent,
+      Repository: context.payload.repository,
+    });
+
+    context.sentry.setExtra('GitHub Payload', {
+      event: context.name,
+      action: context.payload.action,
+      id: context.id,
+      repo: (context.payload.repository) ? context.repo() : undefined,
+    });
+
+    const gitHubInstallationId = context.payload.installation.id;
+    context.log = context.log.child({ gitHubInstallationId });
+
+    if (context.payload.sender.type === 'Bot') {
+      context.log({ noop: 'bot', botId: context.payload.sender.id, botLogin: context.payload.sender.login }, 'Halting futher execution since the sender is a bot');
+      return;
+    }
+
+    if (isFromIgnoredRepo(context.payload)) {
+      context.log({ noop: 'ignored_repo', installation_id: context.payload.installation.id, repository_id: context.payload.repository.id }, 'Halting futher execution since the repository is explicilty ignored');
+      return;
+    }
+
+    // TODO just check the count. Don't fetch all of them.
+    const subscriptions = await Subscription.getAllForInstallation(gitHubInstallationId);
+    if (!subscriptions.length) {
+      context.log({ noop: 'no_subscriptions' }, 'Halting futher execution since no subscriptions were found');
+      return;
+    }
```

Comment:
I am keeping at least one query here in this handler, but it's to check if we have any subscriptions for the installation id. I imagine this will cut down the queuing _considerably_.